src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml

```diff
@@ -278,21 +272,18 @@ let parse_event_from_log_entry ~logger ~network log_entry =
     if
       Result.ok (find bool payload [ "puppeteer_script_event" ])
       |> Option.value ~default:false
-    then
-      let%map msg =
+    then (
+      let%bind msg =
         parse (parser_from_of_yojson Puppeteer_message.of_yojson) payload
       in
-      match msg.event_type with
-      | "node_offline" ->
-          [%log spam] "hitting node_offline event from puppeteer" ;
-          Event_type.Event (Event_type.Node_offline, ())
-      | _ ->
-          failwith "Could not process a puppeteer message from the logs"
+      [%log spam] "parsing puppeteer event" ;
+      Event_type.parse_puppeteer_log msg )
     else
       let%bind msg =
         parse (parser_from_of_yojson Logger.Message.of_yojson) payload
       in
-      Event_type.parse_event msg
+      [%log spam] "parsing daemon event" ;
+      Event_type.parse_daemon_log msg
```

Comment:
@nholland94 you mentioned having a `parse_error`, `parse_daemon_log`, and a `parse_puppeteer_log` but don't get why we would have a `parse_error` function.  we have a different parse function for each input type, but there isn't an error input type, that just happens when the daemon log messes up for some reason.

let me know what you think

Reply:
`parse_error` would be for error logs. The code logically differentiates between structured events in the log and error logs that are not necessarily events. So the logs come from 2 sources, but they essentially get filtered into 3 sources based on the log filtering logic.